Components/WebService/Rest/RestConfigurableProducer.php

```diff
@@ -112,6 +113,7 @@ protected function request(ClientInterface $client, array $stepActionParams, arr
             self::VALIDATION,
             self::REQUEST_QUERY_PARAMETERS,
             self::REQUEST_BASE_URI,
+            self::REQUEST_EXPECTED_RESPONSE_FORMAT
```

Comment:
Couldn't we define the `REQUEST_EXPECTED_RESPONSE_FORMAT` in the same way as `REQUEST_EXPECTED_RESPONSE_TYPE`? I mean
```
$stepParamsResolver->setDefault(self::REQUEST_EXPECTED_RESPONSE_FORMAT, $encoding);
$stepParamsResolver->setDefined([
            RestConfigurableProtocol::OPTION_HEADERS,
            self::VALIDATION,
            self::REQUEST_QUERY_PARAMETERS,
            self::REQUEST_BASE_URI,
])
...
$responseBody = $this->getSerializer()->deserialize(
    $responseContent,
    $params[self::REQUEST_EXPECTED_RESPONSE_TYPE],
    $params[self:: REQUEST_EXPECTED_RESPONSE_FORMAT],
);
```

Reply:
I agree that it would be more elegant but `$encoding` is calculated later on this monstrous method. Getting `$encoding` before this chunk would require rearranging the code a bit and it might have some side effects.

Reply:
Fair enough